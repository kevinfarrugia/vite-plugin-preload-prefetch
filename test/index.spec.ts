import { resolve } from "node:path";

import { format } from "prettier";
import { OutputAsset, RollupOutput } from "rollup";
import { build, InlineConfig } from "vite";
import { describe, expect, test, vi } from "vitest";
import PreloadPrefetch, { PreloadPrefetchOptions } from "../src";
import { configs } from "./fixtures/configs";

async function buildVite(
  options: PreloadPrefetchOptions[],
  inlineConfig: InlineConfig = {}
) {
  const { output } = (await build({
    root: resolve(__dirname, "./fixtures"),
    plugins: [PreloadPrefetch(options)],
    ...inlineConfig,
  })) as RollupOutput;

  const { source: indexSource } = output.find(
    (item) => item.fileName === "index.html"
  ) as OutputAsset;

  return format(indexSource.toString(), { parser: "html" });
}

describe("PreloadPrefetch Plugin", () => {
  Object.keys(configs).forEach((key) => {
    const config = configs[key];
    test(`test ${key}`, async () => {
      const output = await buildVite(config);
      expect(output).toMatchSnapshot();
    });

    test(`test ${key} with basePath`, async () => {
      const output = await buildVite(config, { base: "/base" });
      expect(output).toMatchSnapshot();
    });

    test(`test ${key} with relative basePath `, async () => {
      const output = await buildVite(config, { base: "./" });
      expect(output).toMatchSnapshot();
    });

    test(`test ${key} with URL basePath `, async () => {
      const output = await buildVite(config, {
        base: "https://cdn.example.com/",
      });
      expect(output).toMatchSnapshot();
    });
  });

  test("should warn if no entryMatch or outputMatch is provided", async () => {
    const options: PreloadPrefetchOptions[] = [
      {
        rel: "preload",
        files: [{}], // No entryMatch or outputMatch
        injectTo: "head",
      },
    ];

    const consoleSpy = vi.spyOn(console, "warn").mockImplementation(() => {});
    await buildVite(options);
    expect(consoleSpy).toHaveBeenCalledWith(
      "You should have at least one option between entryMatch and outputMatch."
    );
    consoleSpy.mockRestore();
  });
});
