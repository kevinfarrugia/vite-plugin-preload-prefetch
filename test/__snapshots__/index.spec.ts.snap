// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PreloadPrefetch Plugin > test custom-attributes 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="/assets/index-7FPxZ1Nm.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index-YmWju7QB.css" />
    <link
      rel="preload"
      href="/assets/form-validation-CTJOkw2k.js"
      as="script"
      crossorigin="use-credentials"
    />
    <link
      rel="preload"
      href="/assets/index-7FPxZ1Nm.js"
      as="script"
      crossorigin="use-credentials"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test custom-attributes with URL basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="https://cdn.example.com/assets/index-Cr2LILwB.js"
    ></script>
    <link
      rel="stylesheet"
      crossorigin
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />
    <link
      rel="preload"
      href="https://cdn.example.com/assets/form-validation-CTJOkw2k.js"
      as="script"
      crossorigin="use-credentials"
    />
    <link
      rel="preload"
      href="https://cdn.example.com/assets/index-Cr2LILwB.js"
      as="script"
      crossorigin="use-credentials"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test custom-attributes with basePath 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="/base/assets/index-DoCgpryW.js"
    ></script>
    <link rel="stylesheet" crossorigin href="/base/assets/index-CA6iRNhf.css" />
    <link
      rel="preload"
      href="/base/assets/form-validation-CTJOkw2k.js"
      as="script"
      crossorigin="use-credentials"
    />
    <link
      rel="preload"
      href="/base/assets/index-DoCgpryW.js"
      as="script"
      crossorigin="use-credentials"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test custom-attributes with relative basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="./assets/index-BTUyZHJ8.js"></script>
    <link rel="stylesheet" crossorigin href="./assets/index-DnjSZFrA.css" />
    <link
      rel="preload"
      href="./assets/form-validation-CTJOkw2k.js"
      as="script"
      crossorigin="use-credentials"
    />
    <link
      rel="preload"
      href="./assets/index-BTUyZHJ8.js"
      as="script"
      crossorigin="use-credentials"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test dynamic-imports 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="/assets/index-7FPxZ1Nm.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index-YmWju7QB.css" />
    <link rel="prefetch" href="/assets/form-validation-CTJOkw2k.js" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test dynamic-imports with URL basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="https://cdn.example.com/assets/index-Cr2LILwB.js"
    ></script>
    <link
      rel="stylesheet"
      crossorigin
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />
    <link
      rel="prefetch"
      href="https://cdn.example.com/assets/form-validation-CTJOkw2k.js"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test dynamic-imports with basePath 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="/base/assets/index-DoCgpryW.js"
    ></script>
    <link rel="stylesheet" crossorigin href="/base/assets/index-CA6iRNhf.css" />
    <link rel="prefetch" href="/base/assets/form-validation-CTJOkw2k.js" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test dynamic-imports with relative basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="./assets/index-BTUyZHJ8.js"></script>
    <link rel="stylesheet" crossorigin href="./assets/index-DnjSZFrA.css" />
    <link rel="prefetch" href="./assets/form-validation-CTJOkw2k.js" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-prefetch 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link rel="prefetch" href="/assets/index-YmWju7QB.css" />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="/assets/index-7FPxZ1Nm.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index-YmWju7QB.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-prefetch with URL basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link
      rel="prefetch"
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="https://cdn.example.com/assets/index-Cr2LILwB.js"
    ></script>
    <link
      rel="stylesheet"
      crossorigin
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-prefetch with basePath 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link rel="prefetch" href="/base/assets/index-CA6iRNhf.css" />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="/base/assets/index-DoCgpryW.js"
    ></script>
    <link rel="stylesheet" crossorigin href="/base/assets/index-CA6iRNhf.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-prefetch with relative basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link rel="prefetch" href="./assets/index-DnjSZFrA.css" />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="./assets/index-BTUyZHJ8.js"></script>
    <link rel="stylesheet" crossorigin href="./assets/index-DnjSZFrA.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-preload 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link
      rel="preload"
      href="/assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="/assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="/assets/index-7FPxZ1Nm.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index-YmWju7QB.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-preload with URL basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link
      rel="preload"
      href="https://cdn.example.com/assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="https://cdn.example.com/assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="https://cdn.example.com/assets/index-Cr2LILwB.js"
    ></script>
    <link
      rel="stylesheet"
      crossorigin
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-preload with basePath 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link
      rel="preload"
      href="/base/assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="/base/assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="/base/assets/index-DoCgpryW.js"
    ></script>
    <link rel="stylesheet" crossorigin href="/base/assets/index-CA6iRNhf.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test head-prepend-preload with relative basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <link
      rel="preload"
      href="./assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="./assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="./assets/index-BTUyZHJ8.js"></script>
    <link rel="stylesheet" crossorigin href="./assets/index-DnjSZFrA.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test multiple-rules 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="/assets/index-7FPxZ1Nm.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index-YmWju7QB.css" />
    <link
      rel="preload"
      href="/assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="/assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link rel="prefetch" href="/assets/form-validation-CTJOkw2k.js" />
    <link rel="prefetch" href="/assets/index-7FPxZ1Nm.js" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test multiple-rules with URL basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="https://cdn.example.com/assets/index-Cr2LILwB.js"
    ></script>
    <link
      rel="stylesheet"
      crossorigin
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />
    <link
      rel="preload"
      href="https://cdn.example.com/assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="https://cdn.example.com/assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="prefetch"
      href="https://cdn.example.com/assets/form-validation-CTJOkw2k.js"
    />
    <link
      rel="prefetch"
      href="https://cdn.example.com/assets/index-Cr2LILwB.js"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test multiple-rules with basePath 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="/base/assets/index-DoCgpryW.js"
    ></script>
    <link rel="stylesheet" crossorigin href="/base/assets/index-CA6iRNhf.css" />
    <link
      rel="preload"
      href="/base/assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="/base/assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link rel="prefetch" href="/base/assets/form-validation-CTJOkw2k.js" />
    <link rel="prefetch" href="/base/assets/index-DoCgpryW.js" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test multiple-rules with relative basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="./assets/index-BTUyZHJ8.js"></script>
    <link rel="stylesheet" crossorigin href="./assets/index-DnjSZFrA.css" />
    <link
      rel="preload"
      href="./assets/Roboto-Bold-OBUL28o9.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link
      rel="preload"
      href="./assets/Roboto-Medium-DRylU_ql.woff2"
      type="font/woff2"
      as="font"
      crossorigin="anonymous"
    />
    <link rel="prefetch" href="./assets/form-validation-CTJOkw2k.js" />
    <link rel="prefetch" href="./assets/index-BTUyZHJ8.js" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test no-matching-files 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="/assets/index-7FPxZ1Nm.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index-YmWju7QB.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test no-matching-files with URL basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="https://cdn.example.com/assets/index-Cr2LILwB.js"
    ></script>
    <link
      rel="stylesheet"
      crossorigin
      href="https://cdn.example.com/assets/index-BAgHiCBu.css"
    />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test no-matching-files with basePath 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script
      type="module"
      crossorigin
      src="/base/assets/index-DoCgpryW.js"
    ></script>
    <link rel="stylesheet" crossorigin href="/base/assets/index-CA6iRNhf.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;

exports[`PreloadPrefetch Plugin > test no-matching-files with relative basePath  1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
    <script type="module" crossorigin src="./assets/index-BTUyZHJ8.js"></script>
    <link rel="stylesheet" crossorigin href="./assets/index-DnjSZFrA.css" />
  </head>
  <body>
    <h1>Test App</h1>
    <form id="myForm">
      <input type="text" name="name" value="John Doe" />
      <input type="text" name="age" value="30" />
    </form>
  </body>
</html>
"
`;
